executor/executor.go

```diff
@@ -215,13 +221,33 @@ func Run(path string, env []string, emitter screwdriver.Emitter, build screwdriv
 		return fmt.Errorf("Cannot start shell: %v", err)
 	}
 
+	// Command to Export Env
+	exportEnvCmd :=
+	"prefix='export '; file="+ envFilepath + "; newfile=" + envFilepath + "_export; env > $file && " +
+
+	// Remove PS1, this gives some issues if exporting to ""
+	"sed '/^PS1=.*/d' $file > $newfile && " +
+	"mv $newfile $file && " +
+
+	// Loops through each line
+	"while read -r line; do " +
+	"escapeQuote=`echo $line | sed 's/\"/\\\\\\\"/g'` && " +    //escape double quote
+	"newline=`echo $escapeQuote | sed 's/\\([A-Za-z_][A-Za-z0-9_]*\\)=\\(.*\\)/\\1=\"\\2\"/'` && " +    // add double quote around
+	"echo ${prefix}$newline; " +
+	"done < $file > $newfile"
+
 	// Run setup commands
 	setupCommands := []string{
 		"set -e",
 		"export PATH=$PATH:/opt/sd",
-		"finish() { echo $SD_STEP_ID $?; }",
+		// trap EXIT, echo the last step ID and write ENV to /tmp/buildEnv
+		"finish() { " +
+		"EXITCODE=$?; " +
+		exportEnvCmd + " && " +
```

Comment:
If the exportEnvCmd failed, then this whole cmd will break here and won’t be able to print out the step exit code?